### DISCUSSION TITLE
How to update?

### DISCUSSION THREAD

**Discussion by koiesmith** [Chia-Network/bladebit#89] [Q&A] [answered]
I have several systems now using your program and love it.   Do you have instructions for how to update the app once it's been compiled the first time with your latest code changes?

Also do you frequent discord or anything to chat?  I have systems with 1tb to 2tb of memory I could make available for testing if needed. 

Koie

**Comment by harold-b** [comment, marked as answer]
You can just rebuild after you have pulled the changes.

Inside your local bladebit repo:
```bash
# Switch to the branch you want:
git checkout master

# Grab & merge the remote changes
git pull origin

# Re-build
make clean && make -j$(nproc --all)
```
If you've made no local changes `git pull` will fetch then merge the remote changes to your local branch.  You'll have to merge-in other branches manually if you've already have them checked out locally.

If you've made change you can either discard them by deleting the local branch then pulling or try and merge them, resolving any issues manually.

After you build you can just replace the current executable with the one you just built.

**Reply by harold-b** [reply]
As for the Discord question:

I can be found on the **#plotting** channel on the official Chia Keybase: https://keybase.io/team/chia_network.public

**Reply by koiesmith** [reply]
I can confirm if you are simply a user of bladebit (not modifying code) and have followed the compile instructions originally and need to update to latest code, running these 3 commands from the bladebit directory are all you need to rebuild the application to take advantage of the new changes!
